src/utils/daos_hdlr.c

```diff
@@ -2096,143 +2000,59 @@ file_mkdir(struct cmd_args_s *ap, struct file_dfs *file_dfs,
 			rc = errno;
 		}
 	} else if (file_dfs->type == DAOS) {
-		rc = mkdir_dfs(ap, file_dfs, dir, mode);
+		rc = dfs_sys_mkdir(file_dfs->dfs_sys, dir, *mode, 0);
 		if (rc != 0) {
 			/* mkdir_dfs already prints error */
 			D_GOTO(out, rc);
 		}
 	} else {
 		rc = EINVAL;
-		fprintf(ap->errstream, "File type not known: %s type=%d\n",
-			dir, file_dfs->type);
+		DH_PERROR_SYS(ap, rc, "File type not known '%s' type=%d", dir, file_dfs->type);
 	}
 out:
 	return rc;
 }
 
-static DIR*
-opendir_dfs(struct cmd_args_s *ap, struct file_dfs *file_dfs, const char *dir)
-{
-	int	rc = 0;
-	struct	fs_copy_dirent *dirp;
-
-	D_ALLOC_PTR(dirp);
-	if (dirp == NULL) {
-		errno = ENOMEM;
-		return NULL;
-	}
-	rc = dfs_lookup(file_dfs->dfs, dir, O_RDWR, &dirp->dir, NULL, NULL);
-	if (rc != 0) {
-		DH_PERROR_SYS(ap, rc, "dfs_lookup '%s' failed", dir);
-		errno = rc;
-		D_FREE(dirp);
-	}
-	return (DIR *)dirp;
-}
-
 static DIR*
 file_opendir(struct cmd_args_s *ap, struct file_dfs *file_dfs, const char *dir)
 {
 	DIR *dirp = NULL;
+	int rc	  = 0;
 
 	if (file_dfs->type == POSIX) {
 		dirp = opendir(dir);
+		if (dirp == NULL)
+			rc = errno;
 	} else if (file_dfs->type == DAOS) {
-		dirp = opendir_dfs(ap, file_dfs, dir);
+		rc = dfs_sys_opendir(file_dfs->dfs_sys, dir, 0, &dirp);
+		if (rc != 0)
+			dirp = NULL;
```

Comment:
NULL assignment shouldn't be necessary here, since if the call failed dfs_sys_opendir should/would leave the pointer untouched
```suggestion
		rc = dfs_sys_opendir(file_dfs->dfs_sys, dir, 0, &dirp);
```